File: samples/funcTest0.cpp
#include <iostream>
#include <vector>
using namespace std;

int sum(int x, int y) {
    int result = x + y;
    x++; // x を変化させてみる
    cout << "in sum() " << x << endl;
    return result;
}
void add(int & x, int y) {
    x += y;
    cout << "in add() " << x << endl; 
}

int main(void)
{
    int a = 3;
    int b = sum(a, 4);
    cout << "main0:" << a << "," << b << endl;
    add(a, b);
    cout << "main1:" << a << "," << b << endl;
    return 0;
}

File: samples/func_test0.py
def sum_inc(x, y):
    result = x + y
    x += 1
    print("in sum_inc(), x=" + str(x))
    return result


def main():
    a = 3
    b = sum_inc(a, 4)
    print("in main(): a=" + str(a) 
                 + ", b=" + str(b))


main()
